### PR TITLE
Add COVERALLS_TOKEN plus refactoring

### PR DESCRIPTION
**Pull request by fingolfin**
This contains a rebased version of PR #174 by @simonbyrne plus some refactoring.

Closes #174

### CODE REVIEW

**Comment by simonbyrne** [comment]
What's the Travis failure caused by?

**Comment by fingolfin** [comment]
Cause is me being stupid ;-). The refactored logic for setting `data["repo_token"]` was broken for Travis. I'll try to fix it.

File: src/coveralls.jl [hunk 1]
```diff
@@ -91,6 +85,18 @@ module Coveralls
             error("No compatible CI platform detected")
         end
 
+        repo_token =
+                get(ENV,"COVERALLS_TOKEN") do
+                    get(ENV, "REPO_TOKEN") do #backward compatibility
+                        if data["service_name"] != "travis-ci"
```
Comment (simonbyrne) [inline src/coveralls.jl]: Not a big issue, but do we have any way to detect if it is a private Travis session or not?

**Comment by simonbyrne** [comment]
Thanks!

**Pull request merged**